### ISSUE TITLE
Parameter name does not have appropriate value?

### ISSUE THREAD

**Issue by muteebali** [0x1mason/GribApi.NET#91]
@0x1mason 
By reading multi parametric GRIB file, there are some issues given below, 

1.  Some records have same  **parameter name** as **parameter Id** in numeric form i.e 156,220 etc why is it so?

2.  **Parameter Name** is a concatenated string of _parameter short name_, _parameter_, and _parameter unit_ i.e. **tcc Temperature K**. I have found the value of the **short name** and **unit** explicitly but can not figure out the parameter name explicitly. Is there any property which specifies the parameter name i.e. Temperature, Relative humidity etc. 

3. Is there any parameter to specify the record type i.e. Wind, temperature, Wind direction etc.

**Comment by muteebali** [comment]
@0x1mason 
Can you please answer the above queries.
I would be grateful.
Thanks

**Comment by IdahoSixString** [comment]
@muteebali  
Can you please provide the grib file you are working with?

**Comment by muteebali** [comment]
@IdahoSixString  Here is the file, you can look into it.
Thanks

[isaac.zip](https://github.com/0x1mason/GribApi.NET/files/1363429/isaac.zip)

**Comment by IdahoSixString** [comment]
@0x1mason 
There are a fair amount of unknown parameters in this file. 
@muteebali 
Did you have this set "EnableMultipleFieldMessages = true"?
Looks like there is the U and V components of wind. 
What version of the NuGet are you using? I am using beta4.

**Comment by muteebali** [comment]
@IdahoSixString 
Yes i have already enabled that parameter and also i am using beta 4.

@IdahoSixString @0x1mason 
- Is API is not able to read these unknown parameter or something else? 
- Can we read these unknown parameters ? 
- How to identify specific record, weather it is Wind, Temperature etc. i think so there is not a such a 
   property in extracted data which indicates it?

Thanks

**Comment by muteebali** [comment]
@IdahoSixString  @jnyrup  @0x1mason 
I am sharing some more files, it would be more easy to trace out the problem. These files have specific parametric values. 
I am looking forward to resolve these issues in my project and will be thankful to all of you to help me out.

(https://github.com/0x1mason/GribApi.NET/files/1382340/temperature_max2m.zip)
(https://github.com/0x1mason/GribApi.NET/files/1382341/temperature_min2m.zip)
(https://github.com/0x1mason/GribApi.NET/files/1382342/temperature2m.zip)

**Comment by IdahoSixString** [comment]
@muteebali 
Where are these grib files originally coming from?

**Comment by jnyrup** [comment]
[wgrib](http://www.cpc.ncep.noaa.gov/products/wesley/wgrib.html) seems to have no problem decoding `temperature_max2m.grb`.
It recognizes the data as "CIN=Convective inhibition [J/kg]" and "TMAX=Max. temp. [K]".

Here are the dumps for eccodes and wgrib, respectively. 
[eccodes_log.txt](https://github.com/0x1mason/GribApi.NET/files/1383893/eccodes_log.txt)
[wgrib_log.txt](https://github.com/0x1mason/GribApi.NET/files/1383895/wgrib_log.txt)

**Comment by muteebali** [comment]
Thanks for your response. 

@IdahoSixString 
I have downloaded these GRIB files from [Zygrib](http://www.zygrib.org/)

@jnyrup 
I have gone through these log files and concluded with the following findings:

1. In [eccodes_log.txt](https://github.com/0x1mason/GribApi.NET/files/1385253/eccodes_log.txt) **parameterName**, **parameterUnits** and **indicatorOfParameter** has the value `156`. What does `156` indicate?

2. How do you identify the type of record in **GRIBMessage**? As in `temperature_max2m.grb`, **CIN** and **MaxTemperature** types of record exist.

3. How can i get both **CIN** and **MaxTemperature** values in **GRIBMessage** ? Are CIN and MaxTemperature are treated as separate records in **GRIBMessage**. 

4. How can we generate these log files? 

In my project, I need to get values against cardinal positions. I am unable to identify the type (CIN, Temperature etc.), Unit (J/kg, K etc.) of record in **GRIBMessage**. Can you please share a piece of code to extract records having type and unit for `temperature_max2m.grb`

I am looking forward to your response.

Thanks

**Comment by jnyrup** [comment]
@muteebali 
Just for your information, I am just a consumer of GRIB encoded data as you. 
I'm a programmer and I like to help people, but I really have no expertise in GRIB.

1. I don't know, to me it just seems that eccodes struggles decoding the file.

2, 3: I don't know. At one point I had a problem with GribApi.Net reading multi message files, where wgrib2 seemed to work fine.
My solution back then was to split the files with wgrib2 and then read them with GribApi.NET.
It was hackish and ugly, but it worked. 
If I was in you situation, I would try [compiling EcCodes](https://software.ecmwf.int/wiki/display/ECC/ecCodes+installation) with python interface enabled to debug if it was ecCodes or GribApi.NET that had problems. Please note that I haven't neither tried compiling ecCodes nor tried the python interface, so I can't help you with that.

4. 
For wgrib I used `.\wgrib.exe -v -V temperature_max2m.grb | Out-File -FilePath wgribLog.txt`
For GribApi.NET I used the `GribMessage.Dump()` method to get the parameters.

eccodes should support CIN and TMAX according to their [website](http://apps.ecmwf.int/codes/grib/param-db?filter=grib1). Search for CIN or TMAX.

**Comment by muteebali** [comment]
@jnyrup 
Is there any programmatic way to split files in .NET?

@0x1mason 
Can you please share your view. @jnyrup exactly I am having a problem to read the multi-message file.

**Comment by jnyrup** [comment]
@muteebali The problems I had with multi-message was that I only saw the first message and not the subsequent ones. In fact I'm not really sure if there ever was a problem, I'm suspecting that I just didn't tried enabling `EnableMultipleFieldMessages`.

Your problem seems to be that ecCodes cannot decode your grib1 file, at least its parameters. 
Have you checked if GribApi.NET reads the actual values correct?

I cannot imagine how just splitting you the file into its messages would help ecCodes to read the grib parameters. My suggestion was was more hacky:
Call wgrib from c# via `Process` to split e.g. `temperature_max2m.grb` into 33 files, one for each sub message. Each file would then have its parameters in its filename, e.g `temperature_max2m_CIN_1200_.grb` for the sub message with parameter CIN and forecast 12 hours.
Again this i hacky and ugly, but it gets the job done.

**Comment by 0x1mason** [comment]
This issue was moved to GribApiDotNet/GribApi.NET#4
